src/explore-education-statistics-admin/src/components/editable/EditableKeyStat.tsx

```diff
@@ -1,205 +1,81 @@
-      );
-    }
-
-    const dataDefinitionTitle = summary?.dataDefinitionTitle[0] || 'Help';
-
+  const { updateKeyStatisticDataBlock } = useReleaseContentActions();
+  if ((keyStat as KeyStatisticDataBlock).dataBlockId) {
     return (
-      <>
-        <KeyStatTile
-          title={keyStat.title}
-          value={keyStat.value}
-          testId={testId}
-          isReordering={isReordering}
-        >
-          {summary?.dataSummary[0] && (
-            <p className="govuk-body-s" data-testid={`${testId}-summary`}>
-              {summary.dataSummary[0]}
-            </p>
-          )}
-        </KeyStatTile>
-
-        {summary?.dataDefinition[0] && !isReordering && (
-          <Details
-            summary={dataDefinitionTitle}
-            className={styles.definition}
-            hiddenText={
-              dataDefinitionTitle === 'Help'
-                ? `for ${keyStat.title}`
-                : undefined
-            }
-          >
-            <div data-testid={`${testId}-definition`}>
-              {summary.dataDefinition.map(data => (
-                <ReactMarkdown key={data}>{data}</ReactMarkdown>
-              ))}
-            </div>
-          </Details>
-        )}
-
-        {isEditing && !isReordering && (
-          <ButtonGroup className="govuk-!-margin-top-2">
-            <Button onClick={toggleShowForm.on}>Edit</Button>
+      <EditableKeyStatDataBlock
+        keyStat={keyStat as KeyStatisticDataBlock}
+        testId={testId}
+        isEditing={isEditing}
+        isReordering={isReordering}
+        onRemove={onRemove}
+        onSubmit={async values => {
+          const request: KeyStatisticDataBlockUpdateRequest = {
+            trend: values.trend,
+            guidanceTitle: values.guidanceTitle,
+            guidanceText: values.guidanceText,
+          };
+          await updateKeyStatisticDataBlock({
+            releaseId: keyStat.releaseId,
+            keyStatisticId: keyStat.id,
+            request,
+          });
+        }}
+      />
+    );
+  }
 
-            {onRemove && (
-              <Button
-                disabled={removing}
-                variant="secondary"
-                onClick={() => {
-                  toggleRemoving.on();
-                  onRemove();
-                }}
-              >
-                Remove
-              </Button>
-            )}
-          </ButtonGroup>
-        )}
-      </>
+  if (
+    (keyStat as KeyStatisticText).title &&
+    (keyStat as KeyStatisticText).statistic
+  ) {
+    return (
+      <EditableKeyStatText
+        keyStat={keyStat as KeyStatisticText}
+        testId={testId}
+        isEditing={isEditing}
+        isReordering={isReordering}
+        onRemove={onRemove}
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        onSubmit={async values => {
+          // EES-3913
+        }}
+      />
     );
-  };
+  }
 
-  return <LoadingSpinner loading={isLoading}>{renderInner()}</LoadingSpinner>;
```

Comment:
So we don't need the LoadingSpinner anymore, or will that get reintroduced by one of the following tickets?

Reply:
The `LoadingSpinner` is now in `EditableKeyStatDataBlock`. I think this git diff is misleading as the old `EditableKeyStat` has become `EditableKeyStatDataBlock` - it would make more sense for those files to be diffed. The new `EditableKeyStat` is only determining which type of key stat we have and the appropriate `EditableKeyStat+` component to render.

Reply:
Okeydokie!